app/waf/src/controllers/adc.controller.ts

```diff
@@ -441,25 +471,8 @@ export class AdcController extends BaseController {
       );
 
     switch (Object.keys(actionBody)[0]) {
-      case 'create':
-        if (await this.adcStCtr.readyTo(AdcState.POWERON)) {
-          this.createOn(adc, addonReq);
-          return {id: adc.id};
-        } else
-          throw new HttpErrors.UnprocessableEntity(
-            `Not ready for bigip VE to : ${AdcState.POWERON}`,
-          );
-      case 'delete':
-        if (await this.adcStCtr.readyTo(AdcState.RECLAIMED)) {
-          this.deleteOn(adc, addonReq);
-          return {id: adc.id};
-        } else
-          throw new HttpErrors.UnprocessableEntity(
-            `Not ready for bigip VE to : ${AdcState.RECLAIMED}`,
-          );
-
       case 'setup':
-        if (await this.adcStCtr.readyTo(AdcState.DOINSTALLED)) {
+        if (await this.adcStCtr.readyTo(AdcState.TRUSTED)) {
```

Comment:
Can we also change resource action API to /adcs/{adcId}/setup at this time?

Reply:
This PR is used to  reorganize the API implementation. 
I prefer to make it done in other PRs by someone.

Reply:
Agree.